### DISCUSSION TITLE
Import & export

### DISCUSSION THREAD

**Discussion by cboulanger** [openqda/openqda#142] [General]
Hi, great project! I am delighted to see someone is working towards the goal of getting rid of this mess of proprietary data-silos which we are forced to work with right now. One of the main issues I have with software like NVivo is the lack of any serious form of data interoperability and I envision OpenQDA to shine in this particular area, especially if a community of plugin developers emerges that can contribute import and export adapters. What are your ideas in this area, in particular, are you planning on some sort of plugin architecture?

**Comment by jankapunkt** [comment]
Hi @cboulanger thanks a lot for starting this important topic. We build our whole database schema on top of REFI (https://www.qdasoftware.org/)  which was a joint effort started actually by the big closed vendors for better interoperability (even though many of them still don't fully support it). By doing so we aim to be interoperable by default and as good as possible.

Regarding plugin architecture: yes actually chose this exact pattern! We are currently preparing getting full public, once we chose appropriate licenses etc. and from there teams could theoretically also host this software on-premise and create their own plugins for their own needs.

A publication about the architecture is currently in the works and will be linked here, soon.

**Reply by profkuckuck** [reply]
Hi, we learned about this project at the DGPuK conference and would like to test it with a small scale research project with coding about to  begin in the next weeks because it fits so well to our needs for simple collaboration.

However, given the early state openqda, a possibility to export the project with all the codes would reassure us. I gather that you apply the REFI-QDA standard, but I haven't found any concrete possibility to download data. So I have two very basic and short term questions regarding export: 
- Is exporting our data with categories and codes possible now?
- If not, when will it be possible?

Thank you!

**Reply by jankapunkt** [reply]
Hi @profkuckuck thank you for joining the conversation.

Internally we are close to be "ready" to add export functionality. However, this is still not 100% functional, which is why we added the disclaimer about the data to this early-access release.

We will have meeting this week as a team in order to create a public roadmap, which we will also include here and where we also address our estimate for when import/export will be included.

**Reply by profkuckuck** [reply]
Thanks for the quick reply! 

So we'll stay tuned, and the roadmap will help us decide.

**Reply by jankapunkt** [reply]
@profkuckuck we are now fully public:

Repository: https://github.com/openqda/openqda/

Pre-Release Roadmap: https://github.com/openqda/openqda/milestone/1

Release 1.0 Roadmap: https://github.com/openqda/openqda/milestone/2

**Reply by kadewe** [reply]
and in the pre-release which should be released by the end of the week we have implemented im- and export of codebooks (see roadmap). For Release 1.0 we are going to implement im- and export of full (refi-compatible) projects.

**Comment by jankapunkt** [comment]
@cboulanger we are now fully public:

Repository: https://github.com/openqda/openqda/

Pre-Release Roadmap: https://github.com/openqda/openqda/milestone/1

Release 1.0 Roadmap: https://github.com/openqda/openqda/milestone/2
